Makes census-java compatible with Java 6

sebright opened the pull request:
This PR includes #2, because it uses the Maven build files to build with Java 6 and 7 in .travis.yml.

adriancole commented on core/javatests/com/google/census/CensusScopeTest.java:
```diff
@@ -41,58 +41,76 @@ public void testNoScope() {
   @Test
   public void testScope() {
-    try (CensusScope scope = new CensusScope(Census.getCurrent().with(K1, V1))) {
+    CensusScope scope = new CensusScope(Census.getCurrent().with(K1, V1));
```
not sure if bazel does this, but not really required to limit tests to java6 is it?

sebright replied:
If Census supports Java 6, then I think we should also run the tests with it.  I'd like to add the tests to pom.xml soon.

adriancole commented on core_native/java/com/google/census/CensusSerializer.java:
```diff
@@ -36,14 +34,23 @@ static ByteBuffer serialize(CensusContextImpl context) {
+    try {
+      return ByteBuffer.wrap(builder.toString().getBytes("UTF-8"));
+    } catch (UnsupportedEncodingException e) {
+      throw new AssertionError("UTF-8 not supported.");
```
worthwhile making this a static constant.

```
  public static final Charset UTF_8 = Charset.forName("UTF-8");
```

sebright commented:
I rebased and updated the Maven build files to work with the Protocol Buffers that were introduced in #8.  #8 also removed the use of `StandardCharsets.UTF_8`.
